File: server/src/ApplicationCore/Models/Catalog/ProductDetails.cs
using ApplicationCore.Models.Files;

namespace ApplicationCore.Models.Catalog;

public class ProductDetails
{
    public BrandModel Brand { get; set; }

    public CategoryModel Category { get; set; }

    public string Code { get; set; }

    public decimal Price { get; set; }

    public string Manufacturer { get; set; }

    public int Quantity { get; set; }

    public string Name { get; set; }

    public string ShortDescription { get; set; }

    public string Description { get; set; }

    public string Equipment { get; set; }
    
    public ICollection<CommentModel>? Comments { get; set; }
    
    public ProductRatingModel Rating { get; set; }
    
    public ICollection<CharacteristicModel> Characteristics { get; set; }
    
    public ICollection<Guid>? ProductImagesIds { get; set; }
    
    public ICollection<FileModel>? ProductFiles { get; set; }
}

File: client/src/app/dataModels/catalog/productDetails/productDetails.ts
import { FileModel } from "../../fileModel";
import { BrandModel } from "../brandModel";
import { CategoryModel } from "../categoryModel";
import { CharacteristicModel } from "../characteristicModel";
import { Comment } from "../comment";
import { ProductRating } from "../product/productRating";

export interface ProductDetails {
    name: string;
    code: string;
    description: string;
    brand: BrandModel;
    category: CategoryModel;
    price: number;
    manufacturer: string;
    quantity: number;
    equipment: string;
    rating: ProductRating;
    shortDescription: string;
    characteristics: CharacteristicModel[];
    productImagesIds: string[];
    productFiles: FileModel[];
}